include/dlaf/eigensolver/bt_band_to_tridiag/impl.h

```diff
@@ -983,29 +976,33 @@ void BackTransformationT2B<B, D, T>::call(comm::CommunicatorGrid grid, const Siz
             // W2 = V* . E
             ex::start_detached(
                 dlaf::internal::whenAllLift(blas::Op::ConjTrans, blas::Op::NoTrans, T(1),
-                                            keepFuture(subtile_v), keepFuture(std::move(subtile_e_ro)),
-                                            T(0), splitTile(mat_w2tmp(idx_w2), helper.specW2(nb))) |
+                                            std::move(subtile_v), std::move(subtile_e_ro), T(0),
+                                            splitTile(mat_w2tmp.readwrite(idx_w2), helper.specW2(nb))) |
                 dlaf::tile::gemm(dlaf::internal::Policy<B>(thread_priority::normal)));
 
             // Compute final W2 by adding the contribution from the partner rank
             ex::start_detached(  //
                 comm::scheduleAllSumP2P<B>(mpi_col_comm, rank_partner, tag,
-                                           keepFuture(
-                                               splitTile(mat_w2tmp.read(idx_w2), helper.specW2(nb))),
-                                           splitTile(mat_w2(idx_w2), helper.specW2(nb))));
+                                           splitTile(mat_w2tmp.read(idx_w2), helper.specW2(nb)),
+                                           splitTile(mat_w2.readwrite(idx_w2), helper.specW2(nb))));
 
-            auto subtile_e = splitTile(mat_e_rt(idx_e), spec_e);
+            auto subtile_e = splitTile(mat_e_rt.readwrite(idx_e), spec_e);
             // E -= W . W2
             ex::start_detached(  //
                 dlaf::internal::whenAllLift(blas::Op::NoTrans, blas::Op::NoTrans, T(-1),
-                                            keepFuture(subtile_w),
-                                            keepFuture(splitTile(mat_w2.read(idx_w2), helper.specW2(nb))),
-                                            T(1), std::move(subtile_e)) |
+                                            std::move(subtile_w),
+                                            splitTile(mat_w2.read(idx_w2), helper.specW2(nb)), T(1),
+                                            std::move(subtile_e)) |
                 dlaf::tile::gemm(dlaf::internal::Policy<B>(thread_priority::normal)));
           }
         }
       }
 
+      if (ncols_local == 0) {
```

Comment:
Thanks! Note that this is not just nice to have, but a requirement, since unstarted split senders are not allowed. A quick return is definitely nicer though.